pkg/provider/provider.go

```diff
@@ -225,34 +229,226 @@ func (p *Provider) GetSecretsStoreObjectContent(ctx context.Context, attrib, sec
 	files := []types.SecretFile{}
 	for _, keyVaultObject := range keyVaultObjects {
 		klog.V(5).InfoS("fetching object from key vault", "objectName", keyVaultObject.ObjectName, "objectType", keyVaultObject.ObjectType, "keyvault", mc.keyvaultName, "pod", klog.ObjectRef{Namespace: podNamespace, Name: podName})
-		// fetch the object from Key Vault
-		content, newObjectVersion, err := p.getKeyVaultObjectContent(ctx, kvClient, keyVaultObject, *vaultURL)
+
+		resolvedKvObjects, err := p.resolveObjectVersions(ctx, kvClient, keyVaultObject, *vaultURL)
 		if err != nil {
 			return nil, err
 		}
 
-		objectContent, err := getContentBytes(content, keyVaultObject.ObjectType, keyVaultObject.ObjectEncoding)
+		for _, resolvedKvObject := range resolvedKvObjects {
+			// fetch the object from Key Vault
+			content, newObjectVersion, err := p.getKeyVaultObjectContent(ctx, kvClient, resolvedKvObject, *vaultURL)
+			if err != nil {
+				return nil, err
+			}
+
+			objectContent, err := getContentBytes(content, resolvedKvObject.ObjectType, resolvedKvObject.ObjectEncoding)
+			if err != nil {
+				return nil, err
+			}
+
+			// objectUID is a unique identifier in the format <object type>/<object name>
+			// This is the object id the user sees in the SecretProviderClassPodStatus
+			objectUID := getObjectUID(keyVaultObject.ObjectName, keyVaultObject.ObjectType)
```

Comment:
When we have `objectVersionHistory`, `getObjectUID` would still only return `secret/secret1` instead of all the available files and versions written to the filesystem. Maybe in `getObjectUID()`, we can also pass the `ObjectAlias` and check if this is multiple version. If yes, we append the file suffix `/<suffix idx>` to the uid?

Thoughts @nilekhc @lnr0626?